### PR TITLE
Closes [FEATURE REQUEST] Expand to other vector stores beyond Pinecone

### LINKED ISSUE #65: [FEATURE REQUEST] Expand to other vector stores beyond Pinecone
**Is your feature request related to a problem? Please describe.**
Solely supporting Pinecone is reportedly a deal breaker for some users, see for instance [this comment](https://news.ycombinator.com/item?id=41564857).

**Describe the solution you'd like**
We should expand to other providers, at least top 5 supported by LangChain (Weaviate, Milvus, FAISS, Chroma, Qdrant). We should also allow mixing local LLMs with remote vector stores (currently, local LLMs only go with local Marqo vector store).

### PR DESCRIPTION
**Pull request by LuciAkirami**
# What does this PR do?

I have integrated 4 different vector databases i.e. ChromaDB, FAISS, Milvus, and Qdrant into the existing codebase. I have also edited the `config.py` to choose these vector stores through the command prompt, I have tested each of these by first indexing a repository with these vector stores and chatting with them

Commands used for testing
```bash
sage-index LuciAkirami/liva --no-llm-retriever --mode=remote --embedding-provider=openai --vector-store-provider={replaced with any of the 4 vector database names}
sage-chat LuciAkirami/liva --no-llm-retriever --vector-store-provider={replaced with any of the 4 vector database names} --llm-provider=openai
```

Necessary changes were made to the `pyproject.toml` file to include all the necessary libraries for the above vector stores to work

Closes #65 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag  members/contributors who
may be interested in your PR.

### CODE REVIEW

File: sage/config.py [hunk 1]
```diff
@@ -122,13 +122,38 @@ def add_embedding_args(parser: ArgumentParser) -> Callable:
...
+    parser.add(
+        "--milvus-index-name",
+        default="sage_milvus",
+        help="Milvus collection name. We default it to sage_milvus",
+    )
+    parser.add(
+        "--qdrant-index-name",
+        default="sage_qdrant",
+        help="Qdrant Colleciton name. We default it to sage_qdrant",
+    )
```
Comment (iuliaturc) [inline sage/config.py:151]: Should we collapse all these flags into a single `--index-name`?

File: sage/index.py [hunk 2]
```diff
@@ -41,8 +41,8 @@ def main():
         return
 
     # Additionally validate embedder and vector store compatibility.
-    if args.embedding_provider == "openai" and args.vector_store_provider != "pinecone":
-        parser.error("When using OpenAI embedder, the vector store type must be Pinecone.")
+    if args.embedding_provider == "openai" and args.vector_store_provider not in ["pinecone","chroma","faiss", "milvus", "qdrant"]:
```
Comment (iuliaturc) [inline sage/index.py:44]: Would you mind using a `VectorStoreProvider` enum like we did for [RerankerProvider](https://github.com/Storia-AI/sage/blob/aa5cfeb9981c2d24a127f287658b6e426e7f2dd4/sage/reranker.py#L14)? Then we can get its values via `[member.value for member in VectorStoreProvider]`

File: sage/vector_store.py [hunk 3]
```diff
@@ -191,12 +445,19 @@ def patched_method(self, results):
...
 
-def build_vector_store_from_args(args: dict, data_manager: Optional[DataManager] = None) -> VectorStore:
+def build_vector_store_from_args(args: dict, data_manager: Optional[DataManager] = None,) -> VectorStore:
     """Builds a vector store from the given command-line arguments.
 
     When `data_manager` is specified and hybrid retrieval is requested, we'll use it to fit a BM25 encoder on the corpus
     of documents.
     """
+    if args.embedding_provider == "openai":
+            embeddings = OpenAIEmbeddings(model=args.embedding_model)
```
Comment (iuliaturc) [inline sage/vector_store.py:455]: Nit: remove extra indent

File: sage/vector_store.py [hunk 4]
```diff
@@ -222,6 +483,30 @@ def build_vector_store_from_args(args: dict, data_manager: Optional[DataManager]
             alpha=args.retrieval_alpha,
             bm25_cache=bm25_cache,
         )
+    elif args.vector_store_provider == 'chroma':
+        logging.info("Chroma Store is Being Built")
+        print("ChromaDB is instantiated")
```
Comment (iuliaturc) [inline sage/vector_store.py]: Nit: remove print statement

**Review by iuliaturc — COMMENTED** [review]
Great job! Could you please also run `isort --profile=black && black --line-length=120` for formatting?

File: sage/config.py (hunk 1 above)
Comment (LuciAkirami) [reply sage/config.py:151]: Yeah, that would be good. We can provide a default index name as `sage_index`

File: sage/vector_store.py (hunk 4 above)
Comment (LuciAkirami) [reply sage/vector_store.py]: That's outdated. I have removed it actually

**Comment by LuciAkirami** [comment]
Hello @iuliaturc . Made the required changes as requested

**Review by iuliaturc — APPROVED** [review]
Thank you for your changes! Could you confirm that you've run the pipeline with all of these vector stores and it works?

**Comment by LuciAkirami** [comment]
Yeah, I have tested them, @iuliaturc 

Commands used for testing
```bash
sage-index LuciAkirami/liva --no-llm-retriever --mode=remote --embedding-provider=openai --vector-store-provider={replaced with any of the 4 vector database names}
sage-chat LuciAkirami/liva --no-llm-retriever --vector-store-provider={replaced with any of the 4 vector database names} --llm-provider=openai
```

And I'm able to get relevant responses to the questions asked. The chatbot is able to get the relevant files for the different questions

**Comment by iuliaturc** [comment]
Awesome, thanks so much for your contribution!

**Pull request merged**